{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Goals of this script"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. Import the rMATS data\n",
    "2. Apply a series of cutoffs\n",
    "3. Randomly select a subset of the chosen events and obtain IGV screenshots or rMATS2sashimiplot (this is better and doesn't require manual load)\n",
    "4. Then go through visually and score the events"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Cutoffs to use\n",
    "1. FDR\n",
    "2. dPSI\n",
    "3. Junction Reads\n",
    "4. Gene Expression - Use cuffdiff\n",
    "5. No intervening exons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "pd.set_option('display.max_colwidth', -1)\n",
    "pd.set_option('display.max_columns', None)\n",
    "pd.options.mode.chained_assignment = None  # default='warn'\n",
    "pd.options.mode.chained_assignment = None  # default='warn'\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib_venn as mplv\n",
    "import matplotlib.ticker as ticker\n",
    "import os\n",
    "from math import *\n",
    "from subprocess import *\n",
    "import pybedtools as pbt\n",
    "from glob import glob\n",
    "import seaborn as sns\n",
    "import statsmodels.formula.api as sm\n",
    "import csv\n",
    "import re\n",
    "import scipy.stats as stats\n",
    "import copy\n",
    "csv.register_dialect(\"textdialect\",delimiter='\\t')\n",
    "%matplotlib inline\n",
    "mpl.rc(\"lines\", markeredgewidth=0.5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Master Function\n",
    "def master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi):\n",
    "    \n",
    "    test_df = pd.read_table(rmats_parsing_file)\n",
    "    \n",
    "    print(len(test_df),'number_of_SE_events')\n",
    "    \n",
    "    def test_filter(df,reads,fdr,dpsi,flip_needed):\n",
    "        df = df[df.SAMPLE_1_mean_total > reads]\n",
    "        df = df[df.SAMPLE_2_mean_total > reads]\n",
    "        #now need to generate variable ID and filter based on highest junction read\n",
    "        df['v_id'] = df.splice_id.str.split(':').str[1] + ':' + df.splice_id.str.split(':').str[2] + ':' + df.splice_id.str.split(':').str[5] + \":\" + df.splice_id.str.split(':').str[6] + \":\" + df.GeneID\n",
    "        #make a combo_junctions column that adds up the junctions\n",
    "        df['combo_junctions'] = df.SAMPLE_1_mean_total + df. SAMPLE_2_mean_total\n",
    "        #sort based on these junctions with the highest first\n",
    "        df.sort_values(by='combo_junctions',ascending=False,inplace=True)\n",
    "        #drop duplicates keeping the highest based on variable ID\n",
    "        df.drop_duplicates(subset='v_id',keep='first',inplace=True)\n",
    "        #fdr filter\n",
    "        df = df[df.FDR <= fdr]\n",
    "        #dpsi filter\n",
    "        df = df[df.IncLevelDifference.abs() >= dpsi]\n",
    "        #the flip part is yes if you need to flip the deltaPSI to make positive deltaPSI = promote skipping and negative deltaPSI to promote inclusion\n",
    "        if flip_needed == 'yes':\n",
    "            df['IncLevelDifference'] = df.IncLevelDifference * -1    \n",
    "        return(df)\n",
    "\n",
    "    filt_df = test_filter(test_df,reads,fdr,dpsi,flip_needed)\n",
    "    \n",
    "    print(len(filt_df),'number_of_SE_events_post_reads_fdr_dpsi')\n",
    "    \n",
    "    gene_df = pd.read_table(cuffdiff_file)\n",
    "    gene_df['GeneID'] = gene_df.gene_id.str.split('.').str[0]\n",
    "\n",
    "    #Merge splice and gene\n",
    "    splice_gene_merge = filt_df.merge(gene_df,on='GeneID')\n",
    "    \n",
    "    #filt by FPKM\n",
    "    splice_gene_filt = splice_gene_merge[(splice_gene_merge.value_1 > fpkm)|(splice_gene_merge.value_2 > fpkm)]\n",
    "    \n",
    "    print(len(splice_gene_filt),'number_of_SE_events_post_FPKM')\n",
    "    \n",
    "    gencode_v24 = pbt.BedTool('../annotations/gencode_extracted_regions/prepare_collapsed_exons/gencode.v24lift37.basic_exons.final_collapse.bed')\n",
    "\n",
    "    def filter_out_events_with_intervening_exons(df,exon_bed):\n",
    "        #Get the intron bed\n",
    "        bed_df = df[['chr','upstreamEE','downstreamES','splice_id','IncLevelDifference','strand']]\n",
    "        bed_bt = pbt.BedTool.from_dataframe(bed_df)\n",
    "        #get the exon bed\n",
    "        exon_bed = pbt.BedTool(exon_bed)\n",
    "        #Get the number of whole overlap intersections per event\n",
    "        intersect = bed_bt.intersect(exon_bed,c=True,s=True,F=1)\n",
    "        intersect_names = bed_bt.intersect(exon_bed,wao=True,s=True,F=1)\n",
    "        #make into df\n",
    "        cols = ['chr','upstreamEE','downstreamES','splice_id','IncLevelDifference','strand','num_intersect']\n",
    "        intersect_df = intersect.to_dataframe(names=cols)\n",
    "        cols = ['chr','upstreamEE','downstreamES','splice_id','IncLevelDifference','strand','a','b','c','d','e','f','g']\n",
    "        intersect_names_df = intersect_names.to_dataframe(names=cols)\n",
    "        #collect the ones with only one overlap\n",
    "        one_overlap = intersect_df[intersect_df.num_intersect == 1]\n",
    "        return(intersect_df,intersect_names_df,one_overlap)\n",
    "\n",
    "    if one_variable_only == 'yes':\n",
    "        intersect_df,intersect_names_df,one_overlap = filter_out_events_with_intervening_exons(splice_gene_filt,gencode_v24)\n",
    "        print(len(one_overlap),'number_of_SE_events_with_one_V_exon')\n",
    "    else:\n",
    "        one_overlap = splice_gene_filt\n",
    "    \n",
    "    def run_rmats2sashimiplot(select_df,orig_df,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed):\n",
    "        cols = ['ID','GeneID','geneSymbol','chr','strand','exonStart_0base','exonEnd','upstreamES','upstreamEE','downstreamES','downstreamEE','ID','IJC_SAMPLE_1','SJC_SAMPLE_1','IJC_SAMPLE_2','SJC_SAMPLE_2','IncFormLen','SkipFormLen','PValue','FDR','IncLevel1','IncLevel2','IncLevelDifference']\n",
    "        full_df = orig_df[orig_df.splice_id.isin(select_df.splice_id)]\n",
    "        os.system('mkdir RMATS_VALIDATION/{}'.format(df_name))\n",
    "        \n",
    "        #save the full df\n",
    "        full_df.to_csv('RMATS_VALIDATION/'+df_name+'/'+df_name+'.rmats_filt.full.txt',sep='\\t',index=False)\n",
    "        \n",
    "        #Take a sample percent of these events to make rMATS2SashimiPlot\n",
    "        df = full_df.sample(frac=sample_frac,random_state=sample_seed)\n",
    "\n",
    "        df.to_csv('RMATS_VALIDATION/'+df_name+'/'+df_name+'.rmats_filt.sample.txt',sep='\\t',index=False)\n",
    "        fn = 'RMATS_VALIDATION/'+df_name+'/'+df_name+'.rmats_filt.sample.txt'\n",
    "        if run_sashimi == 'yes':\n",
    "            cmd = 'rmats2sashimiplot --b1 {} --b2 {} -t SE -e {} --l1 {} --l2 {} --exon_s {} --intron_s {} -o {}'.format(bam1_list,bam2_list,fn,label1,label2,exon_scale,intron_scale,'RMATS_VALIDATION/'+df_name)\n",
    "            os.system(cmd)    \n",
    "        \n",
    "    run_rmats2sashimiplot(one_overlap,filt_df,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "ucsc_canon = pbt.BedTool('../annotations/ucsc_known_canonical.bed6')\n",
    "gencode_v24 = pbt.BedTool('../annotations/gencode_extracted_regions/prepare_collapsed_exons/gencode.v24lift37.basic_exons.final_collapse.bed')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Running HTE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Not complex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(58079, 'number_of_SE_events')\n",
      "(528, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(349, 'number_of_SE_events_post_FPKM')\n",
      "(168, 'number_of_SE_events_with_one_V_exon')\n"
     ]
    }
   ],
   "source": [
    "#Remember - these are all gencodev24 basic with rMATS c=0.0001\n",
    "rmats_parsing_file = 'rMATS_parsing_turbo/hte_turbo_loose_all_SE_FDR_1.0_dPSI_0.0_read_cutoff_0.0.txt'\n",
    "cuffdiff_file = '../cuffdiff_runs/HTE_cuffdiff/gene_exp.diff'\n",
    "\n",
    "fdr = 0.05\n",
    "dpsi = 0.15\n",
    "reads = 10\n",
    "fpkm = 5\n",
    "flip_needed = 'yes'\n",
    "one_variable_only = 'yes'\n",
    "run_sashimi = 'yes'\n",
    "\n",
    "exon_scale = 1\n",
    "intron_scale = 4\n",
    "sample_frac = 0.1\n",
    "sample_seed = 10\n",
    "\n",
    "bam1_list = '../star_runs/HTE1Aligned.sortedByCoord.out.bam,../star_runs/HTE3Aligned.sortedByCoord.out.bam'\n",
    "bam2_list = '../star_runs/HTE2Aligned.sortedByCoord.out.bam,../star_runs/HTE4Aligned.sortedByCoord.out.bam'\n",
    "df_name = 'HTE_run1'\n",
    "label1 = 'HTE_WT'\n",
    "label2 = 'HTE_KD'\n",
    "\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Complex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(58079, 'number_of_SE_events')\n",
      "(528, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(349, 'number_of_SE_events_post_FPKM')\n"
     ]
    }
   ],
   "source": [
    "rmats_parsing_file = 'rMATS_parsing_turbo/hte_turbo_loose_all_SE_FDR_1.0_dPSI_0.0_read_cutoff_0.0.txt'\n",
    "cuffdiff_file = '../cuffdiff_runs/HTE_cuffdiff/gene_exp.diff'\n",
    "\n",
    "fdr = 0.05\n",
    "dpsi = 0.15\n",
    "reads = 10\n",
    "fpkm = 5\n",
    "flip_needed = 'yes'\n",
    "one_variable_only = 'no'\n",
    "run_sashimi = 'yes'\n",
    "\n",
    "\n",
    "exon_scale = 1\n",
    "intron_scale = 4\n",
    "sample_frac = 0.1\n",
    "sample_seed = 10\n",
    "\n",
    "bam1_list = '../star_runs/HTE1Aligned.sortedByCoord.out.bam,../star_runs/HTE3Aligned.sortedByCoord.out.bam'\n",
    "bam2_list = '../star_runs/HTE2Aligned.sortedByCoord.out.bam,../star_runs/HTE4Aligned.sortedByCoord.out.bam'\n",
    "df_name = 'HTE_run1_complex'\n",
    "label1 = 'HTE_WT'\n",
    "label2 = 'HTE_KD'\n",
    "\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(76188, 'number_of_SE_events')\n",
      "(780, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(528, 'number_of_SE_events_post_FPKM')\n"
     ]
    }
   ],
   "source": [
    "#original\n",
    "rmats_parsing_file = 'rMATS_parsing_turbo/hte_turbo_original_all_SE_FDR_1.0_dPSI_0.0_read_cutoff_0.0.txt'\n",
    "cuffdiff_file = '../cuffdiff_runs/HTE_cuffdiff/gene_exp.diff'\n",
    "\n",
    "fdr = 0.05\n",
    "dpsi = 0.15\n",
    "reads = 10\n",
    "fpkm = 5\n",
    "flip_needed = 'yes'\n",
    "one_variable_only = 'no'\n",
    "run_sashimi = 'yes'\n",
    "\n",
    "\n",
    "exon_scale = 1\n",
    "intron_scale = 4\n",
    "sample_frac = 0.1\n",
    "sample_seed = 10\n",
    "\n",
    "bam1_list = '../star_runs/HTE1Aligned.sortedByCoord.out.bam,../star_runs/HTE3Aligned.sortedByCoord.out.bam'\n",
    "bam2_list = '../star_runs/HTE2Aligned.sortedByCoord.out.bam,../star_runs/HTE4Aligned.sortedByCoord.out.bam'\n",
    "df_name = 'HTE_run1_original'\n",
    "label1 = 'HTE_WT'\n",
    "label2 = 'HTE_KD'\n",
    "\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(47450, 'number_of_SE_events')\n",
      "(347, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(241, 'number_of_SE_events_post_FPKM')\n"
     ]
    }
   ],
   "source": [
    "#with TSL1 and c = 0.0001\n",
    "rmats_parsing_file = 'rMATS_parsing_turbo/hte_turbo_tsl1_all_SE_FDR_1.0_dPSI_0.0_read_cutoff_0.0.txt'\n",
    "cuffdiff_file = '../cuffdiff_runs/HTE_cuffdiff/gene_exp.diff'\n",
    "\n",
    "fdr = 0.05\n",
    "dpsi = 0.15\n",
    "reads = 10\n",
    "fpkm = 5\n",
    "flip_needed = 'yes'\n",
    "one_variable_only = 'no'\n",
    "run_sashimi = 'yes'\n",
    "\n",
    "exon_scale = 1\n",
    "intron_scale = 4\n",
    "sample_frac = 0.1\n",
    "sample_seed = 10\n",
    "\n",
    "bam1_list = '../star_runs/HTE1Aligned.sortedByCoord.out.bam,../star_runs/HTE3Aligned.sortedByCoord.out.bam'\n",
    "bam2_list = '../star_runs/HTE2Aligned.sortedByCoord.out.bam,../star_runs/HTE4Aligned.sortedByCoord.out.bam'\n",
    "df_name = 'HTE_run1_tsl1'\n",
    "label1 = 'HTE_WT'\n",
    "label2 = 'HTE_KD'\n",
    "\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(58079, 'number_of_SE_events')\n",
      "(263, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(210, 'number_of_SE_events_post_FPKM')\n"
     ]
    }
   ],
   "source": [
    "#With gencode basic, but c = 0.05\n",
    "rmats_parsing_file = 'rMATS_parsing_turbo/hte_turbo_strict_all_SE_FDR_1.0_dPSI_0.0_read_cutoff_0.0.txt'\n",
    "cuffdiff_file = '../cuffdiff_runs/HTE_cuffdiff/gene_exp.diff'\n",
    "\n",
    "fdr = 0.05\n",
    "dpsi = 0.15\n",
    "reads = 10\n",
    "fpkm = 5\n",
    "flip_needed = 'yes'\n",
    "one_variable_only = 'no'\n",
    "run_sashimi = 'yes'\n",
    "\n",
    "exon_scale = 1\n",
    "intron_scale = 4\n",
    "sample_frac = 0.1\n",
    "sample_seed = 10\n",
    "\n",
    "bam1_list = '../star_runs/HTE1Aligned.sortedByCoord.out.bam,../star_runs/HTE3Aligned.sortedByCoord.out.bam'\n",
    "bam2_list = '../star_runs/HTE2Aligned.sortedByCoord.out.bam,../star_runs/HTE4Aligned.sortedByCoord.out.bam'\n",
    "df_name = 'HTE_strict'\n",
    "label1 = 'HTE_WT'\n",
    "label2 = 'HTE_KD'\n",
    "\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>chr</th>\n",
       "      <th>strand</th>\n",
       "      <th>exonStart_0base</th>\n",
       "      <th>exonEnd</th>\n",
       "      <th>upstreamES</th>\n",
       "      <th>upstreamEE</th>\n",
       "      <th>downstreamES</th>\n",
       "      <th>downstreamEE</th>\n",
       "      <th>ID.1</th>\n",
       "      <th>IJC_SAMPLE_1</th>\n",
       "      <th>SJC_SAMPLE_1</th>\n",
       "      <th>IJC_SAMPLE_2</th>\n",
       "      <th>SJC_SAMPLE_2</th>\n",
       "      <th>IncFormLen</th>\n",
       "      <th>SkipFormLen</th>\n",
       "      <th>PValue</th>\n",
       "      <th>FDR</th>\n",
       "      <th>IncLevel1</th>\n",
       "      <th>IncLevel2</th>\n",
       "      <th>IncLevelDifference</th>\n",
       "      <th>splice_id</th>\n",
       "      <th>splicing_direction</th>\n",
       "      <th>IC_SAMPLE_1_mean</th>\n",
       "      <th>SC_SAMPLE_1_mean</th>\n",
       "      <th>SAMPLE_1_mean_total</th>\n",
       "      <th>IC_SAMPLE_2_mean</th>\n",
       "      <th>SC_SAMPLE_2_mean</th>\n",
       "      <th>SAMPLE_2_mean_total</th>\n",
       "      <th>IC_SAMPLE_1_REP1</th>\n",
       "      <th>SC_SAMPLE_1_REP1</th>\n",
       "      <th>IC_SAMPLE_2_REP1</th>\n",
       "      <th>SC_SAMPLE_2_REP1</th>\n",
       "      <th>IC_SAMPLE_1_REP2</th>\n",
       "      <th>SC_SAMPLE_1_REP2</th>\n",
       "      <th>IC_SAMPLE_2_REP2</th>\n",
       "      <th>SC_SAMPLE_2_REP2</th>\n",
       "      <th>IncLevel1_mean</th>\n",
       "      <th>IncLevel2_mean</th>\n",
       "      <th>INC_LEVEL_SAMPLE_1_REP1</th>\n",
       "      <th>INC_LEVEL_SAMPLE_2_REP1</th>\n",
       "      <th>INC_LEVEL_SAMPLE_1_REP2</th>\n",
       "      <th>INC_LEVEL_SAMPLE_2_REP2</th>\n",
       "      <th>v_id</th>\n",
       "      <th>combo_junctions</th>\n",
       "      <th>start</th>\n",
       "      <th>start_1_base</th>\n",
       "      <th>new_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>57993</td>\n",
       "      <td>ENSG00000092841</td>\n",
       "      <td>MYL6</td>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>56554409</td>\n",
       "      <td>56554454</td>\n",
       "      <td>56554026</td>\n",
       "      <td>56554104</td>\n",
       "      <td>56555170</td>\n",
       "      <td>56555334</td>\n",
       "      <td>57993</td>\n",
       "      <td>11445,11646</td>\n",
       "      <td>5721,5224</td>\n",
       "      <td>9599,7361</td>\n",
       "      <td>8815,8173</td>\n",
       "      <td>143</td>\n",
       "      <td>99</td>\n",
       "      <td>7.827072e-14</td>\n",
       "      <td>2.404552e-11</td>\n",
       "      <td>0.581,0.607</td>\n",
       "      <td>0.43,0.384</td>\n",
       "      <td>-0.187</td>\n",
       "      <td>SE:chr12:56554409-56554454:56554026-56554104:56555170-56555334:+:MYL6</td>\n",
       "      <td>skip</td>\n",
       "      <td>11545.5</td>\n",
       "      <td>5472.5</td>\n",
       "      <td>17018.0</td>\n",
       "      <td>8480.0</td>\n",
       "      <td>8494.0</td>\n",
       "      <td>16974.0</td>\n",
       "      <td>11445</td>\n",
       "      <td>5721</td>\n",
       "      <td>9599</td>\n",
       "      <td>8815</td>\n",
       "      <td>11646</td>\n",
       "      <td>5224</td>\n",
       "      <td>7361</td>\n",
       "      <td>8173</td>\n",
       "      <td>0.5940</td>\n",
       "      <td>0.4070</td>\n",
       "      <td>0.581</td>\n",
       "      <td>0.430</td>\n",
       "      <td>0.607</td>\n",
       "      <td>0.384</td>\n",
       "      <td>chr12:56554409-56554454:+:MYL6:ENSG00000092841</td>\n",
       "      <td>33992.0</td>\n",
       "      <td>56554409</td>\n",
       "      <td>56554410</td>\n",
       "      <td>chr12:56554410-56554454</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>34798</td>\n",
       "      <td>ENSG00000060138</td>\n",
       "      <td>YBX3</td>\n",
       "      <td>chr12</td>\n",
       "      <td>-</td>\n",
       "      <td>10862506</td>\n",
       "      <td>10862713</td>\n",
       "      <td>10856621</td>\n",
       "      <td>10856747</td>\n",
       "      <td>10865809</td>\n",
       "      <td>10865932</td>\n",
       "      <td>34798</td>\n",
       "      <td>3876,3562</td>\n",
       "      <td>2874,2614</td>\n",
       "      <td>6313,6427</td>\n",
       "      <td>2778,2238</td>\n",
       "      <td>198</td>\n",
       "      <td>99</td>\n",
       "      <td>7.801011e-10</td>\n",
       "      <td>1.163099e-07</td>\n",
       "      <td>0.403,0.405</td>\n",
       "      <td>0.532,0.589</td>\n",
       "      <td>0.156</td>\n",
       "      <td>SE:chr12:10862506-10862713:10856621-10856747:10865809-10865932:-:YBX3</td>\n",
       "      <td>include</td>\n",
       "      <td>3719.0</td>\n",
       "      <td>2744.0</td>\n",
       "      <td>6463.0</td>\n",
       "      <td>6370.0</td>\n",
       "      <td>2508.0</td>\n",
       "      <td>8878.0</td>\n",
       "      <td>3876</td>\n",
       "      <td>2874</td>\n",
       "      <td>6313</td>\n",
       "      <td>2778</td>\n",
       "      <td>3562</td>\n",
       "      <td>2614</td>\n",
       "      <td>6427</td>\n",
       "      <td>2238</td>\n",
       "      <td>0.4040</td>\n",
       "      <td>0.5605</td>\n",
       "      <td>0.403</td>\n",
       "      <td>0.532</td>\n",
       "      <td>0.405</td>\n",
       "      <td>0.589</td>\n",
       "      <td>chr12:10862506-10862713:-:YBX3:ENSG00000060138</td>\n",
       "      <td>15341.0</td>\n",
       "      <td>10862506</td>\n",
       "      <td>10862507</td>\n",
       "      <td>chr12:10862507-10862713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>27885</td>\n",
       "      <td>ENSG00000084234</td>\n",
       "      <td>APLP2</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>129993506</td>\n",
       "      <td>129993674</td>\n",
       "      <td>129992199</td>\n",
       "      <td>129992408</td>\n",
       "      <td>129996594</td>\n",
       "      <td>129996725</td>\n",
       "      <td>27885</td>\n",
       "      <td>5438,5059</td>\n",
       "      <td>1546,1799</td>\n",
       "      <td>5503,5007</td>\n",
       "      <td>440,385</td>\n",
       "      <td>198</td>\n",
       "      <td>99</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.638,0.584</td>\n",
       "      <td>0.862,0.867</td>\n",
       "      <td>0.254</td>\n",
       "      <td>SE:chr11:129993506-129993674:129992199-129992408:129996594-129996725:+:APLP2</td>\n",
       "      <td>include</td>\n",
       "      <td>5248.5</td>\n",
       "      <td>1672.5</td>\n",
       "      <td>6921.0</td>\n",
       "      <td>5255.0</td>\n",
       "      <td>412.5</td>\n",
       "      <td>5667.5</td>\n",
       "      <td>5438</td>\n",
       "      <td>1546</td>\n",
       "      <td>5503</td>\n",
       "      <td>440</td>\n",
       "      <td>5059</td>\n",
       "      <td>1799</td>\n",
       "      <td>5007</td>\n",
       "      <td>385</td>\n",
       "      <td>0.6110</td>\n",
       "      <td>0.8645</td>\n",
       "      <td>0.638</td>\n",
       "      <td>0.862</td>\n",
       "      <td>0.584</td>\n",
       "      <td>0.867</td>\n",
       "      <td>chr11:129993506-129993674:+:APLP2:ENSG00000084234</td>\n",
       "      <td>12588.5</td>\n",
       "      <td>129993506</td>\n",
       "      <td>129993507</td>\n",
       "      <td>chr11:129993507-129993674</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>23471</td>\n",
       "      <td>ENSG00000155366</td>\n",
       "      <td>RHOC</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>113247721</td>\n",
       "      <td>113247790</td>\n",
       "      <td>113246114</td>\n",
       "      <td>113246428</td>\n",
       "      <td>113249699</td>\n",
       "      <td>113249736</td>\n",
       "      <td>23471</td>\n",
       "      <td>2698,2603</td>\n",
       "      <td>1925,1538</td>\n",
       "      <td>5818,5079</td>\n",
       "      <td>1508,992</td>\n",
       "      <td>167</td>\n",
       "      <td>99</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.454,0.501</td>\n",
       "      <td>0.696,0.752</td>\n",
       "      <td>0.246</td>\n",
       "      <td>SE:chr1:113247721-113247790:113246114-113246428:113249699-113249736:-:RHOC</td>\n",
       "      <td>include</td>\n",
       "      <td>2650.5</td>\n",
       "      <td>1731.5</td>\n",
       "      <td>4382.0</td>\n",
       "      <td>5448.5</td>\n",
       "      <td>1250.0</td>\n",
       "      <td>6698.5</td>\n",
       "      <td>2698</td>\n",
       "      <td>1925</td>\n",
       "      <td>5818</td>\n",
       "      <td>1508</td>\n",
       "      <td>2603</td>\n",
       "      <td>1538</td>\n",
       "      <td>5079</td>\n",
       "      <td>992</td>\n",
       "      <td>0.4775</td>\n",
       "      <td>0.7240</td>\n",
       "      <td>0.454</td>\n",
       "      <td>0.696</td>\n",
       "      <td>0.501</td>\n",
       "      <td>0.752</td>\n",
       "      <td>chr1:113247721-113247790:-:RHOC:ENSG00000155366</td>\n",
       "      <td>11080.5</td>\n",
       "      <td>113247721</td>\n",
       "      <td>113247722</td>\n",
       "      <td>chr1:113247722-113247790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>13505</td>\n",
       "      <td>ENSG00000136238</td>\n",
       "      <td>RAC1</td>\n",
       "      <td>chr7</td>\n",
       "      <td>+</td>\n",
       "      <td>6438292</td>\n",
       "      <td>6438349</td>\n",
       "      <td>6431554</td>\n",
       "      <td>6431672</td>\n",
       "      <td>6439756</td>\n",
       "      <td>6439819</td>\n",
       "      <td>13505</td>\n",
       "      <td>1014,1164</td>\n",
       "      <td>4011,3656</td>\n",
       "      <td>2485,2760</td>\n",
       "      <td>2543,2262</td>\n",
       "      <td>155</td>\n",
       "      <td>99</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.139,0.169</td>\n",
       "      <td>0.384,0.438</td>\n",
       "      <td>0.257</td>\n",
       "      <td>SE:chr7:6438292-6438349:6431554-6431672:6439756-6439819:+:RAC1</td>\n",
       "      <td>include</td>\n",
       "      <td>1089.0</td>\n",
       "      <td>3833.5</td>\n",
       "      <td>4922.5</td>\n",
       "      <td>2622.5</td>\n",
       "      <td>2402.5</td>\n",
       "      <td>5025.0</td>\n",
       "      <td>1014</td>\n",
       "      <td>4011</td>\n",
       "      <td>2485</td>\n",
       "      <td>2543</td>\n",
       "      <td>1164</td>\n",
       "      <td>3656</td>\n",
       "      <td>2760</td>\n",
       "      <td>2262</td>\n",
       "      <td>0.1540</td>\n",
       "      <td>0.4110</td>\n",
       "      <td>0.139</td>\n",
       "      <td>0.384</td>\n",
       "      <td>0.169</td>\n",
       "      <td>0.438</td>\n",
       "      <td>chr7:6438292-6438349:+:RAC1:ENSG00000136238</td>\n",
       "      <td>9947.5</td>\n",
       "      <td>6438292</td>\n",
       "      <td>6438293</td>\n",
       "      <td>chr7:6438293-6438349</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      ID           GeneID geneSymbol    chr strand  exonStart_0base  \\\n",
       "0  57993  ENSG00000092841  MYL6       chr12  +      56554409          \n",
       "1  34798  ENSG00000060138  YBX3       chr12  -      10862506          \n",
       "2  27885  ENSG00000084234  APLP2      chr11  +      129993506         \n",
       "3  23471  ENSG00000155366  RHOC       chr1   -      113247721         \n",
       "4  13505  ENSG00000136238  RAC1       chr7   +      6438292           \n",
       "\n",
       "     exonEnd  upstreamES  upstreamEE  downstreamES  downstreamEE   ID.1  \\\n",
       "0  56554454   56554026    56554104    56555170      56555334      57993   \n",
       "1  10862713   10856621    10856747    10865809      10865932      34798   \n",
       "2  129993674  129992199   129992408   129996594     129996725     27885   \n",
       "3  113247790  113246114   113246428   113249699     113249736     23471   \n",
       "4  6438349    6431554     6431672     6439756       6439819       13505   \n",
       "\n",
       "  IJC_SAMPLE_1 SJC_SAMPLE_1 IJC_SAMPLE_2 SJC_SAMPLE_2  IncFormLen  \\\n",
       "0  11445,11646  5721,5224    9599,7361    8815,8173    143          \n",
       "1  3876,3562    2874,2614    6313,6427    2778,2238    198          \n",
       "2  5438,5059    1546,1799    5503,5007    440,385      198          \n",
       "3  2698,2603    1925,1538    5818,5079    1508,992     167          \n",
       "4  1014,1164    4011,3656    2485,2760    2543,2262    155          \n",
       "\n",
       "   SkipFormLen        PValue           FDR    IncLevel1    IncLevel2  \\\n",
       "0  99           7.827072e-14  2.404552e-11  0.581,0.607  0.43,0.384    \n",
       "1  99           7.801011e-10  1.163099e-07  0.403,0.405  0.532,0.589   \n",
       "2  99           0.000000e+00  0.000000e+00  0.638,0.584  0.862,0.867   \n",
       "3  99           0.000000e+00  0.000000e+00  0.454,0.501  0.696,0.752   \n",
       "4  99           0.000000e+00  0.000000e+00  0.139,0.169  0.384,0.438   \n",
       "\n",
       "   IncLevelDifference  \\\n",
       "0 -0.187                \n",
       "1  0.156                \n",
       "2  0.254                \n",
       "3  0.246                \n",
       "4  0.257                \n",
       "\n",
       "                                                                      splice_id  \\\n",
       "0  SE:chr12:56554409-56554454:56554026-56554104:56555170-56555334:+:MYL6          \n",
       "1  SE:chr12:10862506-10862713:10856621-10856747:10865809-10865932:-:YBX3          \n",
       "2  SE:chr11:129993506-129993674:129992199-129992408:129996594-129996725:+:APLP2   \n",
       "3  SE:chr1:113247721-113247790:113246114-113246428:113249699-113249736:-:RHOC     \n",
       "4  SE:chr7:6438292-6438349:6431554-6431672:6439756-6439819:+:RAC1                 \n",
       "\n",
       "  splicing_direction  IC_SAMPLE_1_mean  SC_SAMPLE_1_mean  SAMPLE_1_mean_total  \\\n",
       "0  skip               11545.5           5472.5            17018.0               \n",
       "1  include            3719.0            2744.0            6463.0                \n",
       "2  include            5248.5            1672.5            6921.0                \n",
       "3  include            2650.5            1731.5            4382.0                \n",
       "4  include            1089.0            3833.5            4922.5                \n",
       "\n",
       "   IC_SAMPLE_2_mean  SC_SAMPLE_2_mean  SAMPLE_2_mean_total  IC_SAMPLE_1_REP1  \\\n",
       "0  8480.0            8494.0            16974.0              11445              \n",
       "1  6370.0            2508.0            8878.0               3876               \n",
       "2  5255.0            412.5             5667.5               5438               \n",
       "3  5448.5            1250.0            6698.5               2698               \n",
       "4  2622.5            2402.5            5025.0               1014               \n",
       "\n",
       "   SC_SAMPLE_1_REP1  IC_SAMPLE_2_REP1  SC_SAMPLE_2_REP1  IC_SAMPLE_1_REP2  \\\n",
       "0  5721              9599              8815              11646              \n",
       "1  2874              6313              2778              3562               \n",
       "2  1546              5503              440               5059               \n",
       "3  1925              5818              1508              2603               \n",
       "4  4011              2485              2543              1164               \n",
       "\n",
       "   SC_SAMPLE_1_REP2  IC_SAMPLE_2_REP2  SC_SAMPLE_2_REP2  IncLevel1_mean  \\\n",
       "0  5224              7361              8173              0.5940           \n",
       "1  2614              6427              2238              0.4040           \n",
       "2  1799              5007              385               0.6110           \n",
       "3  1538              5079              992               0.4775           \n",
       "4  3656              2760              2262              0.1540           \n",
       "\n",
       "   IncLevel2_mean  INC_LEVEL_SAMPLE_1_REP1  INC_LEVEL_SAMPLE_2_REP1  \\\n",
       "0  0.4070          0.581                    0.430                     \n",
       "1  0.5605          0.403                    0.532                     \n",
       "2  0.8645          0.638                    0.862                     \n",
       "3  0.7240          0.454                    0.696                     \n",
       "4  0.4110          0.139                    0.384                     \n",
       "\n",
       "   INC_LEVEL_SAMPLE_1_REP2  INC_LEVEL_SAMPLE_2_REP2  \\\n",
       "0  0.607                    0.384                     \n",
       "1  0.405                    0.589                     \n",
       "2  0.584                    0.867                     \n",
       "3  0.501                    0.752                     \n",
       "4  0.169                    0.438                     \n",
       "\n",
       "                                                v_id  combo_junctions  \\\n",
       "0  chr12:56554409-56554454:+:MYL6:ENSG00000092841     33992.0           \n",
       "1  chr12:10862506-10862713:-:YBX3:ENSG00000060138     15341.0           \n",
       "2  chr11:129993506-129993674:+:APLP2:ENSG00000084234  12588.5           \n",
       "3  chr1:113247721-113247790:-:RHOC:ENSG00000155366    11080.5           \n",
       "4  chr7:6438292-6438349:+:RAC1:ENSG00000136238        9947.5            \n",
       "\n",
       "       start  start_1_base                     new_id  \n",
       "0  56554409   56554410      chr12:56554410-56554454    \n",
       "1  10862506   10862507      chr12:10862507-10862713    \n",
       "2  129993506  129993507     chr11:129993507-129993674  \n",
       "3  113247721  113247722     chr1:113247722-113247790   \n",
       "4  6438292    6438293       chr7:6438293-6438349       "
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "intersection = pd.read_table('../../../systematic_analysis_of_sam_data/hte_rmats_whippet_intersection.txt')\n",
    "intersection.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "hte_orig_parsing = pd.read_table('rMATS_parsing_turbo/hte_turbo_original_all_SE_FDR_1.0_dPSI_0.0_read_cutoff_0.0.txt')\n",
    "hte_new_parsing = pd.read_table('rMATS_parsing_turbo/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "hte_orig_parsing_intersection = hte_orig_parsing[hte_orig_parsing.splice_id.isin(intersection.splice_id)]\n",
    "hte_orig_parsing_intersection.to_csv('rMATS_parsing_turbo/hte_orig_whippet_all_SE.txt',sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>chr</th>\n",
       "      <th>strand</th>\n",
       "      <th>exonStart_0base</th>\n",
       "      <th>exonEnd</th>\n",
       "      <th>upstreamES</th>\n",
       "      <th>upstreamEE</th>\n",
       "      <th>downstreamES</th>\n",
       "      <th>downstreamEE</th>\n",
       "      <th>ID.1</th>\n",
       "      <th>IJC_SAMPLE_1</th>\n",
       "      <th>SJC_SAMPLE_1</th>\n",
       "      <th>IJC_SAMPLE_2</th>\n",
       "      <th>SJC_SAMPLE_2</th>\n",
       "      <th>IncFormLen</th>\n",
       "      <th>SkipFormLen</th>\n",
       "      <th>PValue</th>\n",
       "      <th>FDR</th>\n",
       "      <th>IncLevel1</th>\n",
       "      <th>IncLevel2</th>\n",
       "      <th>IncLevelDifference</th>\n",
       "      <th>splice_id</th>\n",
       "      <th>splicing_direction</th>\n",
       "      <th>IC_SAMPLE_1_mean</th>\n",
       "      <th>SC_SAMPLE_1_mean</th>\n",
       "      <th>SAMPLE_1_mean_total</th>\n",
       "      <th>IC_SAMPLE_2_mean</th>\n",
       "      <th>SC_SAMPLE_2_mean</th>\n",
       "      <th>SAMPLE_2_mean_total</th>\n",
       "      <th>IC_SAMPLE_1_REP1</th>\n",
       "      <th>SC_SAMPLE_1_REP1</th>\n",
       "      <th>IC_SAMPLE_2_REP1</th>\n",
       "      <th>SC_SAMPLE_2_REP1</th>\n",
       "      <th>IC_SAMPLE_1_REP2</th>\n",
       "      <th>SC_SAMPLE_1_REP2</th>\n",
       "      <th>IC_SAMPLE_2_REP2</th>\n",
       "      <th>SC_SAMPLE_2_REP2</th>\n",
       "      <th>IncLevel1_mean</th>\n",
       "      <th>IncLevel2_mean</th>\n",
       "      <th>INC_LEVEL_SAMPLE_1_REP1</th>\n",
       "      <th>INC_LEVEL_SAMPLE_2_REP1</th>\n",
       "      <th>INC_LEVEL_SAMPLE_1_REP2</th>\n",
       "      <th>INC_LEVEL_SAMPLE_2_REP2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>652</td>\n",
       "      <td>ENSG00000101363</td>\n",
       "      <td>MANBAL</td>\n",
       "      <td>chr20</td>\n",
       "      <td>+</td>\n",
       "      <td>35927165</td>\n",
       "      <td>35927282</td>\n",
       "      <td>35918065</td>\n",
       "      <td>35918089</td>\n",
       "      <td>35929610</td>\n",
       "      <td>35929816</td>\n",
       "      <td>652</td>\n",
       "      <td>507,478</td>\n",
       "      <td>161,110</td>\n",
       "      <td>277,216</td>\n",
       "      <td>25,14</td>\n",
       "      <td>198</td>\n",
       "      <td>99</td>\n",
       "      <td>1.735495e-10</td>\n",
       "      <td>2.991491e-08</td>\n",
       "      <td>0.612,0.685</td>\n",
       "      <td>0.847,0.885</td>\n",
       "      <td>-0.217</td>\n",
       "      <td>SE:chr20:35927165-35927282:35918065-35918089:35929610-35929816:+:MANBAL</td>\n",
       "      <td>include</td>\n",
       "      <td>492.5</td>\n",
       "      <td>135.5</td>\n",
       "      <td>628.0</td>\n",
       "      <td>246.5</td>\n",
       "      <td>19.5</td>\n",
       "      <td>266.0</td>\n",
       "      <td>507</td>\n",
       "      <td>161</td>\n",
       "      <td>277</td>\n",
       "      <td>25</td>\n",
       "      <td>478</td>\n",
       "      <td>110</td>\n",
       "      <td>216</td>\n",
       "      <td>14</td>\n",
       "      <td>0.6485</td>\n",
       "      <td>0.8660</td>\n",
       "      <td>0.612</td>\n",
       "      <td>0.847</td>\n",
       "      <td>0.685</td>\n",
       "      <td>0.885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3222</td>\n",
       "      <td>ENSG00000049759</td>\n",
       "      <td>NEDD4L</td>\n",
       "      <td>chr18</td>\n",
       "      <td>+</td>\n",
       "      <td>56002709</td>\n",
       "      <td>56002769</td>\n",
       "      <td>56001049</td>\n",
       "      <td>56001124</td>\n",
       "      <td>56008269</td>\n",
       "      <td>56008401</td>\n",
       "      <td>3222</td>\n",
       "      <td>182,217</td>\n",
       "      <td>502,453</td>\n",
       "      <td>220,239</td>\n",
       "      <td>241,222</td>\n",
       "      <td>158</td>\n",
       "      <td>99</td>\n",
       "      <td>9.099199e-11</td>\n",
       "      <td>1.646674e-08</td>\n",
       "      <td>0.185,0.231</td>\n",
       "      <td>0.364,0.403</td>\n",
       "      <td>-0.175</td>\n",
       "      <td>SE:chr18:56002709-56002769:56001049-56001124:56008269-56008401:+:NEDD4L</td>\n",
       "      <td>include</td>\n",
       "      <td>199.5</td>\n",
       "      <td>477.5</td>\n",
       "      <td>677.0</td>\n",
       "      <td>229.5</td>\n",
       "      <td>231.5</td>\n",
       "      <td>461.0</td>\n",
       "      <td>182</td>\n",
       "      <td>502</td>\n",
       "      <td>220</td>\n",
       "      <td>241</td>\n",
       "      <td>217</td>\n",
       "      <td>453</td>\n",
       "      <td>239</td>\n",
       "      <td>222</td>\n",
       "      <td>0.2080</td>\n",
       "      <td>0.3835</td>\n",
       "      <td>0.185</td>\n",
       "      <td>0.364</td>\n",
       "      <td>0.231</td>\n",
       "      <td>0.403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3948</td>\n",
       "      <td>ENSG00000130714</td>\n",
       "      <td>POMT1</td>\n",
       "      <td>chr9</td>\n",
       "      <td>+</td>\n",
       "      <td>134379575</td>\n",
       "      <td>134379727</td>\n",
       "      <td>134378311</td>\n",
       "      <td>134378460</td>\n",
       "      <td>134381500</td>\n",
       "      <td>134381607</td>\n",
       "      <td>3948</td>\n",
       "      <td>93,96</td>\n",
       "      <td>9,28</td>\n",
       "      <td>158,144</td>\n",
       "      <td>2,3</td>\n",
       "      <td>198</td>\n",
       "      <td>99</td>\n",
       "      <td>5.988043e-08</td>\n",
       "      <td>5.924897e-06</td>\n",
       "      <td>0.838,0.632</td>\n",
       "      <td>0.975,0.96</td>\n",
       "      <td>-0.233</td>\n",
       "      <td>SE:chr9:134379575-134379727:134378311-134378460:134381500-134381607:+:POMT1</td>\n",
       "      <td>include</td>\n",
       "      <td>94.5</td>\n",
       "      <td>18.5</td>\n",
       "      <td>113.0</td>\n",
       "      <td>151.0</td>\n",
       "      <td>2.5</td>\n",
       "      <td>153.5</td>\n",
       "      <td>93</td>\n",
       "      <td>9</td>\n",
       "      <td>158</td>\n",
       "      <td>2</td>\n",
       "      <td>96</td>\n",
       "      <td>28</td>\n",
       "      <td>144</td>\n",
       "      <td>3</td>\n",
       "      <td>0.7350</td>\n",
       "      <td>0.9675</td>\n",
       "      <td>0.838</td>\n",
       "      <td>0.975</td>\n",
       "      <td>0.632</td>\n",
       "      <td>0.960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4831</td>\n",
       "      <td>ENSG00000174151</td>\n",
       "      <td>CYB561D1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>110037141</td>\n",
       "      <td>110037245</td>\n",
       "      <td>110036720</td>\n",
       "      <td>110036909</td>\n",
       "      <td>110037764</td>\n",
       "      <td>110037802</td>\n",
       "      <td>4831</td>\n",
       "      <td>42,32</td>\n",
       "      <td>8,7</td>\n",
       "      <td>51,28</td>\n",
       "      <td>25,21</td>\n",
       "      <td>198</td>\n",
       "      <td>99</td>\n",
       "      <td>1.737617e-03</td>\n",
       "      <td>3.276060e-02</td>\n",
       "      <td>0.724,0.696</td>\n",
       "      <td>0.505,0.4</td>\n",
       "      <td>0.257</td>\n",
       "      <td>SE:chr1:110037141-110037245:110036720-110036909:110037764-110037802:+:CYB561D1</td>\n",
       "      <td>skip</td>\n",
       "      <td>37.0</td>\n",
       "      <td>7.5</td>\n",
       "      <td>44.5</td>\n",
       "      <td>39.5</td>\n",
       "      <td>23.0</td>\n",
       "      <td>62.5</td>\n",
       "      <td>42</td>\n",
       "      <td>8</td>\n",
       "      <td>51</td>\n",
       "      <td>25</td>\n",
       "      <td>32</td>\n",
       "      <td>7</td>\n",
       "      <td>28</td>\n",
       "      <td>21</td>\n",
       "      <td>0.7100</td>\n",
       "      <td>0.4525</td>\n",
       "      <td>0.724</td>\n",
       "      <td>0.505</td>\n",
       "      <td>0.696</td>\n",
       "      <td>0.400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6143</td>\n",
       "      <td>ENSG00000153827</td>\n",
       "      <td>TRIP12</td>\n",
       "      <td>chr2</td>\n",
       "      <td>-</td>\n",
       "      <td>230725121</td>\n",
       "      <td>230725247</td>\n",
       "      <td>230723487</td>\n",
       "      <td>230724290</td>\n",
       "      <td>230744697</td>\n",
       "      <td>230744844</td>\n",
       "      <td>6143</td>\n",
       "      <td>629,655</td>\n",
       "      <td>282,305</td>\n",
       "      <td>1152,924</td>\n",
       "      <td>109,117</td>\n",
       "      <td>198</td>\n",
       "      <td>99</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.527,0.518</td>\n",
       "      <td>0.841,0.798</td>\n",
       "      <td>-0.297</td>\n",
       "      <td>SE:chr2:230725121-230725247:230723487-230724290:230744697-230744844:-:TRIP12</td>\n",
       "      <td>include</td>\n",
       "      <td>642.0</td>\n",
       "      <td>293.5</td>\n",
       "      <td>935.5</td>\n",
       "      <td>1038.0</td>\n",
       "      <td>113.0</td>\n",
       "      <td>1151.0</td>\n",
       "      <td>629</td>\n",
       "      <td>282</td>\n",
       "      <td>1152</td>\n",
       "      <td>109</td>\n",
       "      <td>655</td>\n",
       "      <td>305</td>\n",
       "      <td>924</td>\n",
       "      <td>117</td>\n",
       "      <td>0.5225</td>\n",
       "      <td>0.8195</td>\n",
       "      <td>0.527</td>\n",
       "      <td>0.841</td>\n",
       "      <td>0.518</td>\n",
       "      <td>0.798</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     ID           GeneID geneSymbol    chr strand  exonStart_0base    exonEnd  \\\n",
       "0  652   ENSG00000101363  MANBAL     chr20  +      35927165         35927282    \n",
       "1  3222  ENSG00000049759  NEDD4L     chr18  +      56002709         56002769    \n",
       "2  3948  ENSG00000130714  POMT1      chr9   +      134379575        134379727   \n",
       "3  4831  ENSG00000174151  CYB561D1   chr1   +      110037141        110037245   \n",
       "4  6143  ENSG00000153827  TRIP12     chr2   -      230725121        230725247   \n",
       "\n",
       "   upstreamES  upstreamEE  downstreamES  downstreamEE  ID.1 IJC_SAMPLE_1  \\\n",
       "0  35918065    35918089    35929610      35929816      652   507,478       \n",
       "1  56001049    56001124    56008269      56008401      3222  182,217       \n",
       "2  134378311   134378460   134381500     134381607     3948  93,96         \n",
       "3  110036720   110036909   110037764     110037802     4831  42,32         \n",
       "4  230723487   230724290   230744697     230744844     6143  629,655       \n",
       "\n",
       "  SJC_SAMPLE_1 IJC_SAMPLE_2 SJC_SAMPLE_2  IncFormLen  SkipFormLen  \\\n",
       "0  161,110      277,216      25,14        198         99            \n",
       "1  502,453      220,239      241,222      158         99            \n",
       "2  9,28         158,144      2,3          198         99            \n",
       "3  8,7          51,28        25,21        198         99            \n",
       "4  282,305      1152,924     109,117      198         99            \n",
       "\n",
       "         PValue           FDR    IncLevel1    IncLevel2  IncLevelDifference  \\\n",
       "0  1.735495e-10  2.991491e-08  0.612,0.685  0.847,0.885 -0.217                \n",
       "1  9.099199e-11  1.646674e-08  0.185,0.231  0.364,0.403 -0.175                \n",
       "2  5.988043e-08  5.924897e-06  0.838,0.632  0.975,0.96  -0.233                \n",
       "3  1.737617e-03  3.276060e-02  0.724,0.696  0.505,0.4    0.257                \n",
       "4  0.000000e+00  0.000000e+00  0.527,0.518  0.841,0.798 -0.297                \n",
       "\n",
       "                                                                        splice_id  \\\n",
       "0  SE:chr20:35927165-35927282:35918065-35918089:35929610-35929816:+:MANBAL          \n",
       "1  SE:chr18:56002709-56002769:56001049-56001124:56008269-56008401:+:NEDD4L          \n",
       "2  SE:chr9:134379575-134379727:134378311-134378460:134381500-134381607:+:POMT1      \n",
       "3  SE:chr1:110037141-110037245:110036720-110036909:110037764-110037802:+:CYB561D1   \n",
       "4  SE:chr2:230725121-230725247:230723487-230724290:230744697-230744844:-:TRIP12     \n",
       "\n",
       "  splicing_direction  IC_SAMPLE_1_mean  SC_SAMPLE_1_mean  SAMPLE_1_mean_total  \\\n",
       "0  include            492.5             135.5             628.0                 \n",
       "1  include            199.5             477.5             677.0                 \n",
       "2  include            94.5              18.5              113.0                 \n",
       "3  skip               37.0              7.5               44.5                  \n",
       "4  include            642.0             293.5             935.5                 \n",
       "\n",
       "   IC_SAMPLE_2_mean  SC_SAMPLE_2_mean  SAMPLE_2_mean_total  IC_SAMPLE_1_REP1  \\\n",
       "0  246.5             19.5              266.0                507                \n",
       "1  229.5             231.5             461.0                182                \n",
       "2  151.0             2.5               153.5                93                 \n",
       "3  39.5              23.0              62.5                 42                 \n",
       "4  1038.0            113.0             1151.0               629                \n",
       "\n",
       "   SC_SAMPLE_1_REP1  IC_SAMPLE_2_REP1  SC_SAMPLE_2_REP1  IC_SAMPLE_1_REP2  \\\n",
       "0  161               277               25                478                \n",
       "1  502               220               241               217                \n",
       "2  9                 158               2                 96                 \n",
       "3  8                 51                25                32                 \n",
       "4  282               1152              109               655                \n",
       "\n",
       "   SC_SAMPLE_1_REP2  IC_SAMPLE_2_REP2  SC_SAMPLE_2_REP2  IncLevel1_mean  \\\n",
       "0  110               216               14                0.6485           \n",
       "1  453               239               222               0.2080           \n",
       "2  28                144               3                 0.7350           \n",
       "3  7                 28                21                0.7100           \n",
       "4  305               924               117               0.5225           \n",
       "\n",
       "   IncLevel2_mean  INC_LEVEL_SAMPLE_1_REP1  INC_LEVEL_SAMPLE_2_REP1  \\\n",
       "0  0.8660          0.612                    0.847                     \n",
       "1  0.3835          0.185                    0.364                     \n",
       "2  0.9675          0.838                    0.975                     \n",
       "3  0.4525          0.724                    0.505                     \n",
       "4  0.8195          0.527                    0.841                     \n",
       "\n",
       "   INC_LEVEL_SAMPLE_1_REP2  INC_LEVEL_SAMPLE_2_REP2  \n",
       "0  0.685                    0.885                    \n",
       "1  0.231                    0.403                    \n",
       "2  0.632                    0.960                    \n",
       "3  0.696                    0.400                    \n",
       "4  0.518                    0.798                    "
      ]
     },
     "execution_count": 139,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "check = pd.read_table('rMATS_parsing_turbo/hte_orig_whippet_all_SE.txt')\n",
    "check.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(172, 'number_of_SE_events')\n",
      "(172, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(172, 'number_of_SE_events_post_FPKM')\n"
     ]
    }
   ],
   "source": [
    "#Intersection of rmats original and whippet\n",
    "#Wait - this is also the intersection of new too... doesnt' make sense\n",
    "rmats_parsing_file = 'rMATS_parsing_turbo/hte_orig_whippet_all_SE.txt'\n",
    "cuffdiff_file = '../cuffdiff_runs/HTE_cuffdiff/gene_exp.diff'\n",
    "\n",
    "fdr = 0.05\n",
    "dpsi = 0.15\n",
    "reads = 10\n",
    "fpkm = 5\n",
    "flip_needed = 'yes'\n",
    "one_variable_only = 'no'\n",
    "run_sashimi = 'yes'\n",
    "\n",
    "\n",
    "exon_scale = 1\n",
    "intron_scale = 4\n",
    "sample_frac = 0.1\n",
    "sample_seed = 10\n",
    "\n",
    "bam1_list = '../star_runs/HTE1Aligned.sortedByCoord.out.bam,../star_runs/HTE3Aligned.sortedByCoord.out.bam'\n",
    "bam2_list = '../star_runs/HTE2Aligned.sortedByCoord.out.bam,../star_runs/HTE4Aligned.sortedByCoord.out.bam'\n",
    "df_name = 'HTE_whippet_intersection'\n",
    "label1 = 'HTE_WT'\n",
    "label2 = 'HTE_KD'\n",
    "\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Running HTM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(71051, 'number_of_SE_events')\n",
      "(412, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(288, 'number_of_SE_events_post_FPKM')\n",
      "(140, 'number_of_SE_events_with_one_V_exon')\n"
     ]
    }
   ],
   "source": [
    "rmats_parsing_file = 'rMATS_parsing_turbo/htm_turbo_loose_all_SE_FDR_1.0_dPSI_0.0_read_cutoff_0.0.txt'\n",
    "cuffdiff_file = '../cuffdiff_runs/HTM_cuffdiff/gene_exp.diff'\n",
    "\n",
    "fdr = 0.05\n",
    "dpsi = 0.15\n",
    "reads = 10\n",
    "fpkm = 5\n",
    "flip_needed = 'yes'\n",
    "one_variable_only = 'yes'\n",
    "run_sashimi = 'yes'\n",
    "\n",
    "\n",
    "exon_scale = 1\n",
    "intron_scale = 4\n",
    "sample_frac = 0.1\n",
    "sample_seed = 10\n",
    "\n",
    "bam1_list = '../star_runs/HTM1Aligned.sortedByCoord.out.bam,../star_runs/HTM3Aligned.sortedByCoord.out.bam'\n",
    "bam2_list = '../star_runs/HTM2Aligned.sortedByCoord.out.bam,../star_runs/HTM4Aligned.sortedByCoord.out.bam'\n",
    "df_name = 'HTM_run1'\n",
    "label1 = 'HTM_WT'\n",
    "label2 = 'HTM_KD'\n",
    "\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(71051, 'number_of_SE_events')\n",
      "(412, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(288, 'number_of_SE_events_post_FPKM')\n"
     ]
    }
   ],
   "source": [
    "rmats_parsing_file = 'rMATS_parsing_turbo/htm_turbo_loose_all_SE_FDR_1.0_dPSI_0.0_read_cutoff_0.0.txt'\n",
    "cuffdiff_file = '../cuffdiff_runs/HTM_cuffdiff/gene_exp.diff'\n",
    "\n",
    "fdr = 0.05\n",
    "dpsi = 0.15\n",
    "reads = 10\n",
    "fpkm = 5\n",
    "flip_needed = 'yes'\n",
    "one_variable_only = 'no'\n",
    "run_sashimi = 'yes'\n",
    "\n",
    "\n",
    "exon_scale = 1\n",
    "intron_scale = 4\n",
    "sample_frac = 0.1\n",
    "sample_seed = 10\n",
    "\n",
    "bam1_list = '../star_runs/HTM1Aligned.sortedByCoord.out.bam,../star_runs/HTM3Aligned.sortedByCoord.out.bam'\n",
    "bam2_list = '../star_runs/HTM2Aligned.sortedByCoord.out.bam,../star_runs/HTM4Aligned.sortedByCoord.out.bam'\n",
    "df_name = 'HTM_run1_complex'\n",
    "label1 = 'HTM_WT'\n",
    "label2 = 'HTM_KD'\n",
    "\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Running HTEMT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(67195, 'number_of_SE_events')\n",
      "(811, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(594, 'number_of_SE_events_post_FPKM')\n",
      "(336, 'number_of_SE_events_with_one_V_exon')\n"
     ]
    }
   ],
   "source": [
    "rmats_parsing_file = 'rMATS_parsing_turbo/htemt_turbo_loose_all_SE_FDR_1.0_dPSI_0.0_read_cutoff_0.0.txt'\n",
    "cuffdiff_file = '../cuffdiff_runs/HTEMT_cuffdiff/gene_exp.diff'\n",
    "\n",
    "fdr = 0.05\n",
    "dpsi = 0.15\n",
    "reads = 10\n",
    "fpkm = 5\n",
    "flip_needed = 'no'\n",
    "one_variable_only = 'yes'\n",
    "run_sashimi = 'yes'\n",
    "\n",
    "\n",
    "exon_scale = 1\n",
    "intron_scale = 4\n",
    "sample_frac = 0.1\n",
    "sample_seed = 10\n",
    "\n",
    "bam1_list = '../star_runs/HTE1Aligned.sortedByCoord.out.bam,../star_runs/HTE3Aligned.sortedByCoord.out.bam'\n",
    "bam2_list = '../star_runs/HTM1Aligned.sortedByCoord.out.bam,../star_runs/HTM3Aligned.sortedByCoord.out.bam'\n",
    "df_name = 'HTEMT_run1'\n",
    "label1 = 'HTEMT_EPI'\n",
    "label2 = 'HTEMT_MES'\n",
    "\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(67195, 'number_of_SE_events')\n",
      "(811, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(594, 'number_of_SE_events_post_FPKM')\n"
     ]
    }
   ],
   "source": [
    "rmats_parsing_file = 'rMATS_parsing_turbo/htemt_turbo_loose_all_SE_FDR_1.0_dPSI_0.0_read_cutoff_0.0.txt'\n",
    "cuffdiff_file = '../cuffdiff_runs/HTEMT_cuffdiff/gene_exp.diff'\n",
    "\n",
    "fdr = 0.05\n",
    "dpsi = 0.15\n",
    "reads = 10\n",
    "fpkm = 5\n",
    "flip_needed = 'yes'\n",
    "one_variable_only = 'no'\n",
    "run_sashimi = 'yes'\n",
    "\n",
    "\n",
    "exon_scale = 1\n",
    "intron_scale = 4\n",
    "sample_frac = 0.1\n",
    "sample_seed = 10\n",
    "\n",
    "bam1_list = '../star_runs/HTE1Aligned.sortedByCoord.out.bam,../star_runs/HTE3Aligned.sortedByCoord.out.bam'\n",
    "bam2_list = '../star_runs/HTM1Aligned.sortedByCoord.out.bam,../star_runs/HTM3Aligned.sortedByCoord.out.bam'\n",
    "df_name = 'HTEMT_run1_complex'\n",
    "label1 = 'HTEMT_EPI'\n",
    "label2 = 'HTEMT_MES'\n",
    "\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Testing different cutoffs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#dPSI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(47450, 'number_of_SE_events')\n",
      "(198, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(173, 'number_of_SE_events_post_FPKM')\n",
      "\n",
      "\n",
      "(47450, 'number_of_SE_events')\n",
      "(163, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(139, 'number_of_SE_events_post_FPKM')\n",
      "\n",
      "\n",
      "(47450, 'number_of_SE_events')\n",
      "(107, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(88, 'number_of_SE_events_post_FPKM')\n"
     ]
    }
   ],
   "source": [
    "#What should I test for one sample set? Let's see how much we get from an example set HTE\n",
    "rmats_parsing_file = 'rMATS_parsing_turbo/hte_turbo_tsl1_c05_all_SE_FDR_1.0_dPSI_0.0_read_cutoff_0.0.txt'\n",
    "cuffdiff_file = '../cuffdiff_runs/HTE_cuffdiff/gene_exp.diff'\n",
    "\n",
    "fdr = 0.05\n",
    "dpsi = 0.1\n",
    "reads = 10\n",
    "fpkm = 5\n",
    "cutoff_string = '_'.join([str(fdr),str(dpsi),str(reads),str(fpkm)])\n",
    "\n",
    "flip_needed = 'yes'\n",
    "one_variable_only = 'no'\n",
    "run_sashimi = 'no'\n",
    "\n",
    "exon_scale = 1\n",
    "intron_scale = 4\n",
    "sample_frac = 0.1\n",
    "sample_seed = 10\n",
    "\n",
    "bam1_list = '../star_runs/HTE1Aligned.sortedByCoord.out.bam,../star_runs/HTE3Aligned.sortedByCoord.out.bam'\n",
    "bam2_list = '../star_runs/HTE2Aligned.sortedByCoord.out.bam,../star_runs/HTE4Aligned.sortedByCoord.out.bam'\n",
    "df_name = 'HTE_loose_'+cutoff_string\n",
    "label1 = 'HTE_WT'\n",
    "label2 = 'HTE_KD'\n",
    "\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)\n",
    "print('\\n')\n",
    "dpsi = 0.15\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)\n",
    "print('\\n')\n",
    "dpsi = 0.2\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#FDR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(47450, 'number_of_SE_events')\n",
      "(224, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(190, 'number_of_SE_events_post_FPKM')\n",
      "\n",
      "\n",
      "(47450, 'number_of_SE_events')\n",
      "(198, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(173, 'number_of_SE_events_post_FPKM')\n",
      "\n",
      "\n",
      "(47450, 'number_of_SE_events')\n",
      "(156, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(140, 'number_of_SE_events_post_FPKM')\n"
     ]
    }
   ],
   "source": [
    "#What should I test for one sample set? Let's see how much we get from an example set HTE\n",
    "rmats_parsing_file = 'rMATS_parsing_turbo/hte_turbo_tsl1_c05_all_SE_FDR_1.0_dPSI_0.0_read_cutoff_0.0.txt'\n",
    "cuffdiff_file = '../cuffdiff_runs/HTE_cuffdiff/gene_exp.diff'\n",
    "\n",
    "fdr = 0.1\n",
    "dpsi = 0.1\n",
    "reads = 10\n",
    "fpkm = 5\n",
    "cutoff_string = '_'.join([str(fdr),str(dpsi),str(reads),str(fpkm)])\n",
    "\n",
    "flip_needed = 'yes'\n",
    "one_variable_only = 'no'\n",
    "run_sashimi = 'no'\n",
    "\n",
    "exon_scale = 1\n",
    "intron_scale = 4\n",
    "sample_frac = 0.1\n",
    "sample_seed = 10\n",
    "\n",
    "bam1_list = '../star_runs/HTE1Aligned.sortedByCoord.out.bam,../star_runs/HTE3Aligned.sortedByCoord.out.bam'\n",
    "bam2_list = '../star_runs/HTE2Aligned.sortedByCoord.out.bam,../star_runs/HTE4Aligned.sortedByCoord.out.bam'\n",
    "df_name = 'HTE_loose_'+cutoff_string\n",
    "label1 = 'HTE_WT'\n",
    "label2 = 'HTE_KD'\n",
    "\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)\n",
    "print('\\n')\n",
    "fdr = 0.05\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)\n",
    "print('\\n')\n",
    "fdr = 0.01\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#JUNCTION"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(47450, 'number_of_SE_events')\n",
      "(224, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(190, 'number_of_SE_events_post_FPKM')\n",
      "\n",
      "\n",
      "(47450, 'number_of_SE_events')\n",
      "(217, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(189, 'number_of_SE_events_post_FPKM')\n",
      "\n",
      "\n",
      "(47450, 'number_of_SE_events')\n",
      "(200, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(184, 'number_of_SE_events_post_FPKM')\n"
     ]
    }
   ],
   "source": [
    "#What should I test for one sample set? Let's see how much we get from an example set HTE\n",
    "rmats_parsing_file = 'rMATS_parsing_turbo/hte_turbo_tsl1_c05_all_SE_FDR_1.0_dPSI_0.0_read_cutoff_0.0.txt'\n",
    "cuffdiff_file = '../cuffdiff_runs/HTE_cuffdiff/gene_exp.diff'\n",
    "\n",
    "fdr = 0.1\n",
    "dpsi = 0.1\n",
    "reads = 10\n",
    "fpkm = 5\n",
    "cutoff_string = '_'.join([str(fdr),str(dpsi),str(reads),str(fpkm)])\n",
    "\n",
    "flip_needed = 'yes'\n",
    "one_variable_only = 'no'\n",
    "run_sashimi = 'no'\n",
    "\n",
    "exon_scale = 1\n",
    "intron_scale = 4\n",
    "sample_frac = 0.1\n",
    "sample_seed = 10\n",
    "\n",
    "bam1_list = '../star_runs/HTE1Aligned.sortedByCoord.out.bam,../star_runs/HTE3Aligned.sortedByCoord.out.bam'\n",
    "bam2_list = '../star_runs/HTE2Aligned.sortedByCoord.out.bam,../star_runs/HTE4Aligned.sortedByCoord.out.bam'\n",
    "df_name = 'HTE_loose_'+cutoff_string\n",
    "label1 = 'HTE_WT'\n",
    "label2 = 'HTE_KD'\n",
    "\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)\n",
    "print('\\n')\n",
    "reads = 20\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)\n",
    "print('\\n')\n",
    "reads = 40\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#FPKM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(47450, 'number_of_SE_events')\n",
      "(198, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(188, 'number_of_SE_events_post_FPKM')\n",
      "\n",
      "\n",
      "(47450, 'number_of_SE_events')\n",
      "(198, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(173, 'number_of_SE_events_post_FPKM')\n",
      "\n",
      "\n",
      "(47450, 'number_of_SE_events')\n",
      "(198, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(145, 'number_of_SE_events_post_FPKM')\n"
     ]
    }
   ],
   "source": [
    "#What should I test for one sample set? Let's see how much we get from an example set HTE\n",
    "rmats_parsing_file = 'rMATS_parsing_turbo/hte_turbo_tsl1_c05_all_SE_FDR_1.0_dPSI_0.0_read_cutoff_0.0.txt'\n",
    "cuffdiff_file = '../cuffdiff_runs/HTE_cuffdiff/gene_exp.diff'\n",
    "\n",
    "fdr = 0.05\n",
    "dpsi = 0.1\n",
    "reads = 10\n",
    "fpkm = 3\n",
    "cutoff_string = '_'.join([str(fdr),str(dpsi),str(reads),str(fpkm)])\n",
    "\n",
    "flip_needed = 'yes'\n",
    "one_variable_only = 'no'\n",
    "run_sashimi = 'no'\n",
    "\n",
    "exon_scale = 1\n",
    "intron_scale = 4\n",
    "sample_frac = 0.1\n",
    "sample_seed = 10\n",
    "\n",
    "bam1_list = '../star_runs/HTE1Aligned.sortedByCoord.out.bam,../star_runs/HTE3Aligned.sortedByCoord.out.bam'\n",
    "bam2_list = '../star_runs/HTE2Aligned.sortedByCoord.out.bam,../star_runs/HTE4Aligned.sortedByCoord.out.bam'\n",
    "df_name = 'HTE_loose_'+cutoff_string\n",
    "label1 = 'HTE_WT'\n",
    "label2 = 'HTE_KD'\n",
    "\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)\n",
    "print('\\n')\n",
    "fpkm = 5\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)\n",
    "print('\\n')\n",
    "fpkm = 10\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(47450, 'number_of_SE_events')\n",
      "(198, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(173, 'number_of_SE_events_post_FPKM')\n",
      "\n",
      "\n",
      "(47450, 'number_of_SE_events')\n",
      "(198, 'number_of_SE_events_post_reads_fdr_dpsi')\n",
      "(173, 'number_of_SE_events_post_FPKM')\n",
      "(106, 'number_of_SE_events_with_one_V_exon')\n"
     ]
    }
   ],
   "source": [
    "#One Variable\n",
    "#What should I test for one sample set? Let's see how much we get from an example set HTE\n",
    "rmats_parsing_file = 'rMATS_parsing_turbo/hte_turbo_tsl1_c05_all_SE_FDR_1.0_dPSI_0.0_read_cutoff_0.0.txt'\n",
    "cuffdiff_file = '../cuffdiff_runs/HTE_cuffdiff/gene_exp.diff'\n",
    "\n",
    "fdr = 0.05\n",
    "dpsi = 0.1\n",
    "reads = 10\n",
    "fpkm = 5\n",
    "cutoff_string = '_'.join([str(fdr),str(dpsi),str(reads),str(fpkm)])\n",
    "\n",
    "flip_needed = 'yes'\n",
    "one_variable_only = 'no'\n",
    "run_sashimi = 'no'\n",
    "\n",
    "exon_scale = 1\n",
    "intron_scale = 4\n",
    "sample_frac = 0.1\n",
    "sample_seed = 10\n",
    "\n",
    "bam1_list = '../star_runs/HTE1Aligned.sortedByCoord.out.bam,../star_runs/HTE3Aligned.sortedByCoord.out.bam'\n",
    "bam2_list = '../star_runs/HTE2Aligned.sortedByCoord.out.bam,../star_runs/HTE4Aligned.sortedByCoord.out.bam'\n",
    "df_name = 'HTE_loose_'+cutoff_string\n",
    "label1 = 'HTE_WT'\n",
    "label2 = 'HTE_KD'\n",
    "\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)\n",
    "print('\\n')\n",
    "one_variable_only = 'yes'\n",
    "master_rmats_validation(rmats_parsing_file,cuffdiff_file,df_name,bam1_list,bam2_list,label1,label2,exon_scale,intron_scale,sample_frac,sample_seed,flip_needed,fdr,dpsi,reads,one_variable_only,run_sashimi)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
